lib/sql.js

```diff
@@ -1159,8 +1159,9 @@ SQLConnector.prototype._buildWhere = function(model, where) {
  * @returns {string} The ORDER BY clause
  */
 SQLConnector.prototype.buildOrderBy = function(model, order) {
+  var idName = this.idName(model);
```

Comment:
I think the right fix is to change the assertion at `https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/basic-querying.test.js#L488` to check if the result either is `[0, 1]` or `[1, 0]` as there is no guarantee that a DB will return matching records in the order of `inq` keys.